### PR TITLE
Improve confidence method in LoadMcStas by using descriptor 

### PR DESCRIPTION
**Pull request by Anders-Markvardsen**
Implements [#11141](http://trac.mantidproject.org/mantid/ticket/11141)

To tester: Check unit tests passing and the confidence() method makes use of a descriptor.

PS: 2nd commit with title "AutoTestData now External Data" is not part of #11141, but have added to correct a separate commit where a file in AutoTestData had been added in.

### CODE REVIEW

**Comment by martyngigg** [comment]
Jenkins, retest this please.

**Comment by martyngigg** [comment]
Could you update the title of the pull request to use be a short descriptive sentence rather than the branch name?

File: Code/Mantid/Framework/DataHandling/src/LoadMcStas.cpp [hunk 1]
```diff
@@ -543,24 +543,24 @@ void LoadMcStas::readHistogramData(
  */
 int LoadMcStas::confidence(Kernel::NexusDescriptor &descriptor) const {
   using namespace ::NeXus;
```
Comment (KarlPalmen) [inline Code/Mantid/Framework/DataHandling/src/LoadMcStas.cpp]: This function returns a number indicating its confidence that the file being loaded is a McStas file.

Comment (Anders-Markvardsen) [inline Code/Mantid/Framework/DataHandling/src/LoadMcStas.cpp]: A comment to test a return comment to Karl's comment.

(Found this functionality did not work in one version of internet explorer!)

File: Code/Mantid/Framework/DataHandling/src/LoadMcStas.cpp [hunk 2]
```diff
@@ -543,24 +543,24 @@ void LoadMcStas::readHistogramData(
...
-      auto firstIt = entries.begin();
-      file.openGroup(firstIt->first, firstIt->second);
-      file.openGroup("simulation", "NXnote");
-      std::string nameAttrValue;
-      file.readData("name", nameAttrValue);
-      if (boost::iequals(nameAttrValue, "mccode"))
-        confidence = 98;
-      file.closeGroup();
-      file.closeGroup();
+  if(descriptor.pathExists("/entry1/simulation/name")) {
```
Comment (KarlPalmen) [inline Code/Mantid/Framework/DataHandling/src/LoadMcStas.cpp]: This will speed up loading of any non-McStas file.

**Pull request merged**